tests/decorators/tests.py

```diff
@@ -466,17 +604,583 @@ def a_view(request):
+        @sensitive_post_parameters()
+        async def an_async_view(request):
+            return HttpResponse()
+        req = HttpRequest()
+        resp = await an_async_view(req)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(req.sensitive_post_parameters, '__ALL__')
+
+
+class GzipDecoratorsTests(TestCase):
+    """
+    Tests for the gzip decorator.
+    """
+    # Gzip ignores content that is too short
+    content = "Content " * 100
+
+    def test_gzip_decorator(self):
+        @gzip_page
+        def a_view(request):
+            return HttpResponse(content=self.content)
+        req = HttpRequest()
+        req.META['HTTP_ACCEPT_ENCODING'] = 'gzip'
+        resp = a_view(req)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.headers['Content-Encoding'], 'gzip')
+
+    async def test_gzip_decorator_with_async_view(self):
+        @gzip_page
+        async def an_async_view(request):
+            return HttpResponse(content=self.content)
+        req = HttpRequest()
+        req.META['HTTP_ACCEPT_ENCODING'] = 'gzip'
+        resp = await an_async_view(req)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.headers['Content-Encoding'], 'gzip')
+
+
+class VaryDecoratorsTests(TestCase):
+    """
+    Tests for the vary decorator.
+    """
+    def test_vary_on_headers_decorator(self):
+        @vary_on_headers('Header', 'Another-header')
+        def a_view(request):
+            return HttpResponse()
+        resp = a_view(HttpRequest())
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn('Header, Another-header', resp.headers['Vary'])
+
+    async def test_vary_on_headers_decorator_with_async_view(self):
+        @vary_on_headers('Header', 'Another-header')
+        async def an_async_view(request):
+            return HttpResponse()
+        resp = await an_async_view(HttpRequest())
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn('Header, Another-header', resp.headers['Vary'])
+
+    def test_vary_on_cookie_decorator(self):
+        @vary_on_cookie
+        def a_view(request):
+            return HttpResponse()
+        resp = a_view(HttpRequest())
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn('Cookie', resp.headers['Vary'])
+
+    async def test_vary_on_cookie_decorator_with_async_view(self):
+        @vary_on_cookie
+        async def an_async_view(request):
+            return HttpResponse()
+        resp = await an_async_view(HttpRequest())
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn('Cookie', resp.headers['Vary'])
+
+
+class ConditionalDecoratorTests(TestCase):
+    """
+    Tests for the conditional decorators.
+    """
+    def etag_func(self, request, *args, **kwargs):
+        return '"abc123"'
+
+    def last_modified_func(self, request, *args, **kwargs):
+        return datetime(2020, 1, 1, 0, 0, 0)
```

Comment:
I think `datetime(2020, 1, 1)` is the same thing, i.e. the rest of the zeros default.